.github/workflows/golangci-lint.yaml

```diff
@@ -0,0 +1,55 @@
+name: golangci-lint
+on:
+  push:
+    branches:
+      - main
+  pull_request:
```

Comment:
@bennyz shouldn't it run on pull_request?

Reply:
I think it will run, but yes, it would make sense to remove push here